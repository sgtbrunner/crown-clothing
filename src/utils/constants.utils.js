// INITIAL STATE
export const DEFAULT_INITIAL_INPUT_FIELD_STATE = {
  isValid: false,
  value: '',
  showError: false,
};

// FORM STRINGS
export const DISPLAY_NAME = 'displayName';
export const EMAIL = 'email';
export const PASSWORD = 'password';
export const CONFIRM_PASSWORD = 'confirmPassword';

// ERROR MESSAGES
export const DISPLAY_NAME_ERROR_MESSAGE =
  'Display name must have at least 3 characters';
export const EMAIL_ERROR_MESSAGE = 'Please enter a valid email address';
export const PASSWORD_ERROR_MESSAGE =
  'Your password must be at least 6 characters long';
export const CONFIRM_PASSWORD_ERROR_MESSAGE =
  'Password and confirm password do not match';
export const UNKNOWN_ERROR =
  'An unknown error occurred. Please try again later!';

// OTHER
export const EMPTY_CART_MESSAGE = 'Your cart is empty...';
export const BUTTON = 'button';
