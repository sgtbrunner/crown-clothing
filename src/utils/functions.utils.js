const EMAIL_REGEX_KEY = /^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i;

export const capitalizeFirstLetter = (string) =>
  string[0].toUpperCase() + string.slice(1);

export const isEmailValid = (email) =>
  EMAIL_REGEX_KEY.test(String(email).toLowerCase());
