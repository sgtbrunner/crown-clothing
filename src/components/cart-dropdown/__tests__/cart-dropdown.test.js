import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import { EMPTY_CART_MESSAGE, BUTTON } from '../../../utils/constants.utils';
import CartDropdown from '../cart-dropdown.component';

jest.mock('react-router-dom', () => ({
  withRouter: (component) => component,
}));

const mockStore = configureMockStore();
let store = mockStore({
  cart: {
    cartItems: [],
  },
});

const newCartItems = [
  {
    id: 1,
    price: 10,
    name: 'Item 1',
    quantity: 2,
    imageUrl: '',
  },
  {
    id: 2,
    price: 20,
    name: 'Item 2',
    quantity: 3,
    imageUrl: '',
  },
];

const props = {
  history: {
    push: jest.fn(),
  },
  dispatch: jest.fn(),
};

const ITEM = 'item';
const CHECKOUT = 'CHECKOUT';

const mountComponent = () =>
  render(
    <Provider store={store}>
      <CartDropdown {...props} />
    </Provider>
  );

describe('CartDropdown component', () => {
  it('should render component snapshot', () => {
    const { container } = mountComponent();

    expect(container).toMatchSnapshot();
  });

  it('should render the empty cart message if cart holds no items', () => {
    mountComponent();

    const cartDropdownMessage = screen.getByText(EMPTY_CART_MESSAGE);
    const cartDropdownItems = screen.queryAllByAltText(ITEM);

    expect(cartDropdownMessage).toBeInTheDocument();
    expect(cartDropdownItems.length).toBe(0);
  });

  it('should render 2 (two) items on the dropdown if cart holds 2 (two) items', () => {
    store = mockStore({
      cart: {
        cartItems: newCartItems,
      },
    });

    mountComponent();

    const cartDropdownMessage = screen.queryByText(EMPTY_CART_MESSAGE);
    const cartDropdownItems = screen.getAllByAltText(ITEM);

    expect(cartDropdownMessage).not.toBeInTheDocument();
    expect(cartDropdownItems).toHaveLength(2);
  });

  it('should render the checkout button', () => {
    mountComponent();

    const checkoutButton = screen.getByRole(BUTTON, { name: CHECKOUT });
    expect(checkoutButton).toBeInTheDocument();
  });

  it('should redirect to checkout page on checkout button click', () => {
    mountComponent();

    const checkoutButton = screen.getByRole(BUTTON, { name: CHECKOUT });
    userEvent.click(checkoutButton);

    expect(props.history.push).toHaveBeenCalled();
  });
});
