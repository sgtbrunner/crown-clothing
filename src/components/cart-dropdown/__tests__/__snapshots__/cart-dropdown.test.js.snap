// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartDropdown component should render component snapshot 1`] = `
<div>
  <div
    class="sc-eCApnc bAPMlK"
  >
    <div
      class="sc-jSFjdj hUiUQX"
    >
      <span
        class="sc-gKAaRy kcLyYT"
      >
        Your cart is empty...
      </span>
    </div>
    <button
      class="sc-hKFxyN bAKAQB"
    >
      CHECKOUT
    </button>
  </div>
</div>
`;
